### ISSUE TITLE
Grab Button & Live Mob Interactivity

### ISSUE THREAD

**Issue by ShotCosine** [haubna/PhysicsMod#273]
While Haubna is working hard implementing our favorite features, I have come up with a revolutionary new feature idea that could create a whole new layer of immersion and interactivity in the mod. I present the **Grab Button** - a customizable button that lets users interact with dynamic block physics and ragdolls!

Think Skyrim - how you can pick up objects and throw them around with a different button than the "interact" one.
- All particles and dynamic effects added to the mod are client side for the most part, which means grabbing them and throwing them around wouldn't affect serverside compatibility.
- I am also aware of some users' distaste in adding new controls to the game - especially those with hundreds of mods like me. Maybe instead of adding a whole new control to the game, right-click could be mapped to also grab objects.
- Using a button that's different than right click could be more useful however, imagine peeking through a door without opening it or dragging a ragdoll while at the same time opening doors and using a crafting table.
- A setting could be added to either hold the button to interact or toggle to pick up/put down, maybe even hold to throw.

Another feature that this idea made me think of was **live-mob-interactivity**, which would essentially be grabbing a zombie's arm or a spider's leg and moving it around, all client-sided of course. 
- This would have a similar effect to an animation mod, except the "animations" would be dynamic and interactable - kind of like door and trapdoor physics.
- Swords and arrows could bounce a mob's limb or head back a little bit depending on where it was hit, and maybe could cause dismemberment if ragdolls are enabled. This would only be possible if the mod could detect where the mob was hit from though.
- Only default mobs would be affected just like with ragdolls by default, maybe others could be configured on or off. It seems like most modded mobs break apart on death with "blocky" physics, so maybe these "blocks" or limbs could be detected even if it was modded.
- Limbs that are connected to two or more other limbs wouldn't be affected to avoid moving around the torso of a mob.
- Pushing a mob would cause all it's limbs to bounce around to add to the effect.

In conclusion, the Grab Button definitely feels like a main feature waiting to be added. I'm sure it would be pretty easy to add in and to implement to existing features. Think of all the possibilities! Live Mob Interactivity seems more challenging, but would blow every animation mod out of the water and add to the immersion of the mod. I'm pretty sure those animation mods aren't compatible anyways. Open for discussion!

**Comment by NoshaY2212** [comment]
BIG BRAIN

**Comment by QAQlinmiao** [comment]
#150

**Comment by SargRReal** [comment]
actually, this is a great idea

**Comment by haubna** [comment]
Closed due to moving feature suggestions to the [discord](https://discord.gg/VsNs9xP) (channel #feature-suggestions)
